src/UploadArtifactTask.cpp

```diff
@@ -15,11 +15,13 @@ namespace connect
 
 namespace prc = prism::connect;
 
-Status UploadBackgroundTask::execute(ArtifactUploadHelper* uploader) const
+Status UploadBackgroundTask::execute(ClientSession& session) const
 {
-    return uploader->uploadBackground(timestamp_,
-                                      image_->isFile() ? Payload(image_->getFilePath())
-                                        : Payload(image_->getData(), image_->getDataSize(), image_->getMimeType()));
+    return session.client.uploadBackground(
+                session.accountId, session.cameraId, timestamp_,
+                image_->isFile() ? Payload(image_->getFilePath())
```

Comment:
How about making a separate function for making `Payload` from `image_` instead of repeating that 3 times?